packages/template-retail-react-app/app/components/dynamic-image/index.test.js

```diff
@@ -97,16 +112,81 @@ describe('Dynamic Image Component', () => {
+        })
+
+        test('renders an image with explicit widths', () => {
+            const {getByTestId, getAllByTitle} = renderWithProviders(
+                <DynamicImage
+                    data-testid={'dynamic-image'}
+                    src={src}
+                    imageProps={{
+                        ...imageProps,
+                        loading: 'lazy'
+                    }}
+                    widths={['50vw', '50vw', '20vw', '20vw', '25vw']}
+                />
+            )
+
+            const wrapper = getByTestId('dynamic-image')
+            const elements = getAllByTitle(imageProps.title)
+            expect(elements).toHaveLength(1)
+            expect(elements[0]).toHaveAttribute('src', src)
+            expect(elements[0]).toHaveAttribute('loading', 'lazy')
+            expect(elements[0]).toHaveAttribute('decoding', 'async')
+            expect(elements[0]).not.toHaveAttribute('sizes')
+            expect(elements[0]).not.toHaveAttribute('srcset')
+
+            expect(wrapper.firstElementChild).not.toBe(elements[0])
+            expect(wrapper.firstElementChild.tagName.toLowerCase()).toBe('picture')
+
+            const sourceElements = Array.from(wrapper.querySelectorAll('source'))
+            expect(sourceElements).toHaveLength(5)
+            expect(sourceElements[0]).toHaveAttribute('media', '(min-width: 80em)')
+            expect(sourceElements[0]).toHaveAttribute('sizes', '25vw')
+            expect(sourceElements[0]).toHaveAttribute(
+                'srcset',
+                [384, 768].map((width) => `${src} ${width}w`).join(', ')
+            )
+            expect(sourceElements[1]).toHaveAttribute('media', '(min-width: 62em)')
+            expect(sourceElements[1]).toHaveAttribute('sizes', '20vw')
+            expect(sourceElements[1]).toHaveAttribute(
+                'srcset',
+                [256, 512].map((width) => `${src} ${width}w`).join(', ')
+            )
+            expect(sourceElements[2]).toHaveAttribute('media', '(min-width: 48em)')
+            expect(sourceElements[2]).toHaveAttribute('sizes', '20vw')
+            expect(sourceElements[2]).toHaveAttribute(
+                'srcset',
+                [198, 396].map((width) => `${src} ${width}w`).join(', ')
+            )
+            expect(sourceElements[3]).toHaveAttribute('media', '(min-width: 30em)')
+            expect(sourceElements[3]).toHaveAttribute('sizes', '50vw')
+            expect(sourceElements[3]).toHaveAttribute(
+                'srcset',
+                [384, 768].map((width) => `${src} ${width}w`).join(', ')
+            )
+            expect(sourceElements[4]).not.toHaveAttribute('media')
+            expect(sourceElements[4]).toHaveAttribute('sizes', '50vw')
+            expect(sourceElements[4]).toHaveAttribute(
+                'srcset',
+                [240, 480].map((width) => `${src} ${width}w`).join(', ')
+            )
+
+            expect(Helmet.peek()?.linkTags ?? []).toStrictEqual([])
```

Comment:
Not seeing an explicit test for the error thrown by invalid component usage, e.g.

```
 <DynamicImage
                    data-testid={'dynamic-image'}
                    src={src}
                    imageProps={{
                        ...imageProps,
                        loading: 'lazy'
                    }}
                    widths={['INVALID']}
                />
                ```

Reply:
Hmm, in that case it wasn't there before either. Didn't add/change anything to the widths logic/handling.